consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/headervalidationrules/NoDifficultyRule.java

```diff
@@ -39,11 +39,17 @@ public boolean validate(
       LOG.warn("unable to get total difficulty, parent {} not found", header.getParentHash());
       return false;
     }
-    if (totalDifficulty
-        .get()
-        .greaterOrEqualThan(
-            protocolContext.getConsensusContext(MergeContext.class).getTerminalTotalDifficulty())) {
-      return (header.getDifficulty() == null || header.getDifficulty().equals(Difficulty.ZERO));
+    if (protocolContext.getConsensusContext(MergeContext.class).isPostMerge()) {
+      if (totalDifficulty
+          .get()
+          .greaterThan(
```

Comment:
yes, i think you are right. This should be saying if the current chain has reached TTD, evaluate passed in block as a PoS block...

Which makes me think we have another problem, and this will break if we ever re-validate a past block after the chain has merged.